nolibc/src/lib.rs

```diff
@@ -18,6 +18,30 @@ pub use x86_64::*;
 
 // TODO: These functions are naive implementations that are not performant.
 
+/// Buffer used by readv() and writev() syscalls
+#[repr(C)]
+pub struct Iovec<'a, T> {
+    /// Buffer start address
+    pub base: *mut u8,
+
+    /// Number of bytes to transfer
+    pub size: usize,
+
+    phantom: core::marker::PhantomData<&'a T>,
+}
```

Comment:
```suggestion
#[repr(C)]
pub struct Iovec<'a> {
    /// Buffer start address
    pub base: *mut u8,

    /// Number of bytes to transfer
    pub size: usize,

    phantom: core::marker::PhantomData<&'a ()>,
}
```

Reply:
I didn't realize one could do this. Seems much better!